hw/ip/pattgen/doc/_index.md

```diff
@@ -0,0 +1,143 @@
+Each Channel consists of one clock (`pcl`) and one data (`pda`) line.
+The output channels may be activated and operated independently, or they can be started at the same time to effectively create a 4-output pattern.
+ 
+## Compatibility
+
+This IP block does not have any direct hardware compatibility requirements.
+
+# Theory of Operations 
+
+To start pattern generation, the registers interface of PATTGEN HWIP should be properly initialized and configured.
+To configure a single channel:
+1. Before configuration, disable the desired channel by clearing the corresponding enable bit: ({{< "regref PATT_CTRL.ENABLE_CH0" >}} for Channel 0, {{< regref "PATT_CTRL.ENABLE_CH1" >}} for Channel 1).
+1. Set the polarity bit ({{< regref "PATT_CTRL.POLARITY_CH0" >}} for Channel 0, {{< regref "PATT_CTRL.POLARITY_CH1" >}} for Channel 1) to determine the desired clock phase.
+For either channel, a zero in the polarity bit indicates that the channel clock line (`pcl`) should start low, and the channel data line `pda` transition on every falling edge of `pcl`.
+A one in the polarity bit inverts the `pcl` clock so that it starts high and `pda` transitions on the rising edge.
+{{<wavejson>}}
+{signal: [
+  {name: 'PATT_CTRL.ENABLE_CH0', wave: 'lh......'},
+  {name: 'PATT_CTRL.POLARITY_CH0 (default: low)', wave: '0.......'},
+  {name: 'pcl0_tx', wave: '0.hlhlhl'},
+  {name: 'pda0_tx', wave: 'x3.3.3.3', data: 'DATA[0] DATA[1] DATA[2]'},
+  {name: 'PATT_CTRL.POLARITY_CH1 (high)', wave: '1.......'},
+  {name: 'pcl1_tx', wave: '1.lhlhlh'},
+  {name: 'pda1_tx', wave: 'x5.5.5.5', data: 'DATA[0] DATA[1] DATA[2]'},
+  
+],
+  head: {text: 'Effect of the Polarity Registers',tick:0}}
+{{</wavejson>}}
+
+1. Program the seed patterns (1-64 bits in length) using the multi-registers {{< regref "PATT_DATA_CH0" >}} (for Channel 0) or {{<regref "PATT_CH1_DATA" >}} (for Channel 1).
+For either channel, the first 32 bits to be transmitted, are programmed in the lower half of the multi-register (e.g. {{< regref "PATT_DATA_CH0_0" >}}), and the latter 32 bits are programmed in the upper half of multi-register (e.g. {{< regref "PATT_DATA_CH0_1" >}}).
+1. Program the clock divider ratios to the appropriate registers, {{< regref "PATT_PREDIV_CH0.CLK_RATIO" >}} for Channel 0 or {{< regref "PATT_PREDIV_CH1.CLK_RATIO" >}} for Channel 1.
+The resulting clock frequency will be slower than the input I/O clock by a ratio of 2&times;(CLK_RATIO+1):
+$$f_{pclx}=\frac{f_\textrm{I/O clk}}{2(\textrm{CLK_RATIO}+1)}$$
+1. Program the length of seed pattern using one of the fields {{< regref "PATT_SIZE.LEN_CH0" >}}, for Channel 0. or {{< regref "PATT_SIZE.LEN_CH1" >}} for Channel 1.
+Note that since the allowed seed length ranges from 1-64, the value of this field should be one less than the pattern length.
+For example, to generate an 16-bit pattern on Channel 1, a value of 15 should be written to the field {{< regref "PATT_SIZE.LEN_CH1" >}}. 
```

Comment:
It probably doesn't make a difference to the hardware, but as a programmer I'd first program the length, and then the data. Can you move this bullet point up before the "Program the seed patterns" bullet point?

Reply:
I suppose it doesn't matter. I can change this.

Reply:
Done.